tests/base_scraper/test_base_scraper.py

```diff
@@ -74,5 +74,39 @@ def test_notimplementedexceptions(self):
         self.assertRaises(mozdownload.NotImplementedError,
                           scraper.build_filename, 'invalid binary')
 
+    def test_authentication(self):
+        """testing authentication"""
+        username = 'mozilla'
+        password = 'mozilla'
+        basic_auth_url = 'http://mozqa.com/data/mozqa.com/http_auth/basic/'
+        no_auth_url = 'https://ci.mozilla.org/' #does not need auth
```

Comment:
I would propose that you split this test method into two different ones. Means one for basic auth, and the other for optional basic auth. Right now code is a bit mixed and makes it harder to understand.